pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -239,6 +239,18 @@ def _map_tool_result_part(
                 )
             else:
                 assert_never(resource)
+        elif isinstance(part, mcp_types.ResourceLink):
+            resource_result: mcp_types.ReadResourceResult = anyio.run(lambda: self._client.read_resource(part.uri))
+            content = resource_result.contents[0]
```

Comment:
The [docs](https://modelcontextprotocol.io/docs/concepts/resources) suggest that we should handle all contents, not just number 0:

> Servers may return multiple resources in response to one resources/read request. This could be used, for example, to return a list of files inside a directory when the directory is read.

So if there are more one item in the list, we should process and return them as a list.